android/src/main/java/com/expensify/livemarkdown/RangeSplitter.java

```diff
@@ -0,0 +1,44 @@
+package com.expensify.livemarkdown;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RangeSplitter {
+  public static void splitRangesOnEmojis(List<MarkdownRange> markdownRanges, String type) {
+    List<MarkdownRange> emojiRanges = new ArrayList<>();
+    List<MarkdownRange> oldRanges = new ArrayList<>(markdownRanges);
+    markdownRanges.clear();
+    for (MarkdownRange range : oldRanges) {
+      if (range.getType().equals("emoji")) {
+        emojiRanges.add(range);
+      }
+    }
+
+    int i = 0;
+    int j = 0;
+    while (i < oldRanges.size()) {
+      MarkdownRange currentRange = oldRanges.get(i);
+      if (!currentRange.getType().equals(type)) {
+        markdownRanges.add(currentRange);
+        i += 1;
+        continue;
+      }
+
+      // Split range
```

Comment:
Let's provide a better explanation of what is done here, the method is called `splitRangesOnEmojis` so this comments isn't very helpful.